rmw_cyclonedds_cpp/src/serdata.cpp

```diff
@@ -574,6 +582,49 @@ uint32_t sertype_rmw_hash(const struct ddsi_sertype * tpcmn)
   return h1 ^ h2;
 }
 
+size_t sertype_get_serialized_size(const struct ddsi_sertype * d, const void * sample)
+{
+  const struct sertype_rmw * type = static_cast<const struct sertype_rmw *>(d);
+  try {
+    // ROS 2 doesn't do keys, so its all data (?)
+    if (!type->is_request_header) {
+      return type->cdr_writer->get_serialized_size(sample);
+    } else {
+      // inject the service invocation header data into the CDR stream
+      auto wrap = *static_cast<const cdds_request_wrapper_t *>(sample);
```

Comment:
So for request headers we somehow cannot take the sample here and have to modify the pointer type, dereference it to get some wrapper `wrap` and ask for its serialized size?

OK, but I do not know whether this is correct.

One would think this could be done internally by `get_serialized_size` as it should have type information due to `type->cdr_writer`.